File: gaia-sdk-python/gaia_sdk/graphql/request/input/ConnectSetNodeImpulse.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class ConnectSetNodeImpulse():
    """
    The specification to set an edge for a node
    """
    type: EdgeType
    target: str
    properties: dict
    weight: float

    def __init__(self, type: EdgeType, target: str, properties: dict, weight: float):
        self.type = type
        self.target = target
        self.properties = properties
        self.weight = weight

    def __eq__(self, other):
        if type(other) is type(self):
            return self.type == other.type and self.target == other.target and self.properties == other.properties and self.weight == other.weight
        return False

    def __repr__(self):
        return {'type': self.type, 'target': self.target, 'properties': self.properties, 'weight': self.weight}

File: gaia-sdk-javascript/src/graphql/request/input/ConnectSetNodeImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * The specification to set an edge for a node
 */
export class ConnectSetNodeImpulse {
public _typeName = "ConnectSetNodeImpulse";
    private type:EdgeType;
    private target:Uuid;
    private properties:Struct;
    private weight:Number;

    constructor (type:EdgeType, target:Uuid, properties:Struct, weight:Number) {
        this.type = type;
        this.target = target;
        this.properties = properties;
        this.weight = weight;
    }
}
